features/cellular/TARGET_DRAGONFLY_MODEM/dragonfly_modem_driver/DragonFlyCellularInterface.h

```diff
@@ -0,0 +1,239 @@
+/* Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _DRAGONFLY_CELLULAR_INTERFACE_
+#define _DRAGONFLY_CELLULAR_INTERFACE_
+
+#include "mbed.h"
+#include "nsapi.h"
+#include "rtos.h"
+#include "FileHandle.h"
+#include "InterruptIn.h"
+#include "ATParser.h"
+
+// Forward declaration
+class NetworkStack;
```

Comment:
I'd say "can't you just declare it rather than pull in unneeded headers"? 

Anyone actually needing/using the network stack must be getting it from another file - one of the Socket.h, I imagine.

(There's certainly some code messing with interfaces but not using sockets or the network stack - that easy connect thing in the mbed client examples.)